markets/perps-market/contracts/storage/AsyncOrder.sol

```diff
@@ -348,11 +350,18 @@ library AsyncOrder {
             revert InsufficientMargin(runtime.currentAvailableMargin, runtime.totalRequiredMargin);
         }
 
-        int256 lockedCreditDelta = perpsMarketData.requiredCreditForSize(
-            MathUtil.abs(runtime.newPositionSize).toInt() - MathUtil.abs(oldPosition.size).toInt(),
-            PerpsPrice.Tolerance.DEFAULT
-        );
-        GlobalPerpsMarket.load().validateMarketCapacity(lockedCreditDelta);
+        /// @custom:magnitude determines if more market credit is required
+        /// when a position's magnitude is increased, more credit is required
+        /// when a position's magnitude is decreased, less credit is required
+        uint newMagnitude = MathUtil.abs(runtime.newPositionSize);
+        uint oldMagnitude = MathUtil.abs(oldPosition.size);
+        if (newMagnitude > oldMagnitude) {
```

Comment:
i think a bit more than that please, they need to be lower magnitude and have same side, to be able to skip the validation on min credit.
note the tricky part is that 0 is same side as both long and short

Reply:
To reiterate, is someone flips from 10 eth long to 9 eth short. this is basically allowed since magnitude works, but i would prefer that the guy closes completely rather than continue trading... since the market does not have enough lp to back it.

Reply:
`MathUtil.isSameSideReducing`